### PR TITLE
fix: show 200 status code for successful queries

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

[This change](https://github.com/grafana/loki/pull/10956/files#diff-cee1265b2b7d985c9e33bebaa2299074a017697246ec6c7ada151ccf2f45b499R246) in #10956 broke the status code reporting in the `metrics.go` line from our queriers. In the case where error was `nil`, it produces a 500. Previously, when [error was nil, it produced a 200](https://github.com/grafana/loki/pull/10956/files#diff-71063175d4d67c35bfdd623b3609a70c6667416c640196d78931e0f2f5c0af0eL99).

This PR fixes that behavior, reporting a 200 status unless there is a problem.

### CODE REVIEW

File: cmd/loki/loki-local-config.yaml [hunk 1]
```diff
@@ -18,9 +18,6 @@ common:
     kvstore:
       store: inmemory
 
-ingester_rf1:
```
Comment (trevorwhitney) [inline cmd/loki/loki-local-config.yaml:21]: this appears to have been removed from the code, but was not removed from the config? I got an error on `main` when trying to use this config.

**Review by ashwanthgoli — APPROVED** [review]
LGTM

i opened up a [similar pr](https://github.com/grafana/loki/pull/12102) a while back, but failed to merge it. I'll close it once this is merged.

**Comment by trevorwhitney** [comment]
@ashwanthgoli I like yours better, closing in favor of https://github.com/grafana/loki/pull/12102

**Comment by trevorwhitney** [comment]
nevermind, we still need it in `engine.go`

**Comment by trevorwhitney** [comment]
nevermind again, I can't read

**Pull request closed without merging**